lib/dry/schema/message_compiler.rb

```diff
@@ -116,6 +116,7 @@ def visit_unexpected_key(node, _opts)
 
         Message.new(
           path: path,
+          meta: msg[:meta] || {},
```

Comment:
```suggestion
          meta: msg[:meta] || EMPTY_HASH,
```

This way we won't be allocating extra hash. Message objects are "immutable" so this is perfectly fine.